crates/ruff_linter/src/rules/pylint/rules/bad_super_call.rs

```diff
@@ -0,0 +1,130 @@
+use ruff_python_ast::{
+    Expr, ExprAttribute, ExprCall, ExprName, Stmt, StmtClassDef, StmtExpr, StmtFunctionDef,
+};
+
+use crate::checkers::ast::Checker;
+
+/// ### What it does
+/// Checks to see another argument other than the current class is given as the first argument of the super builtin
+///
+/// ### Why is this bad?
+/// In Python 2.7, `super()` has to be called with its own class and `self`
+/// as arguments `(super(Cat, self))`, which can lead to a mix up of parent and child class in the code.
+///
+/// In Python 3 the recommended way is to call `super()` without arguments
+/// (see also [`super-with-arguments`](https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/super-with-arguments.html)).
+///
+/// ## Problematic code
+/// ```python
+/// class Animal:
+///     pass
+///
+///
+/// class Tree:
+///     pass
+///
+///
+/// class Cat(Animal):
+///     def __init__(self):
+///         super(Tree, self).__init__()  # [bad-super-call]
+///         super(Animal, self).__init__()
+/// ```
+///
+/// ## Correct code
+/// ```python
+/// class Animal:
+///     pass
+///    
+///  
+/// class Tree:
+///     pass
+///    
+///    
+/// class Cat(Animal):
+///     def __init__(self):
+///         super(Animal, self).__init__()
+/// ```
+///
+/// ## References
+/// - [Python documentation: `class super`](https://docs.python.org/3/library/functions.html#super)
+#[violation]
+pub struct BadSuperCall {
+    bad_super_arg: String,
+}
+
+impl Violation for BadSuperCall {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let Self { bad_super_arg } = self;
+        format!("Bad first argument '{bad_super_arg}' given to super()")
+    }
+}
+
+fn is_bad_super_call<'a>(first_arg: &'a Expr, current_class_name: &str) -> Option<&'a str> {
+    let Expr::Name(ExprName { id, .. }) = first_arg else {
+        return None;
+    };
+    (id != current_class_name).then_some(id)
+}
+
+fn check_expr(expr: &Expr, current_class_name: &str) -> Option<Diagnostic> {
+    if let Expr::Call(ExprCall {
+        range,
+        func,
+        arguments,
+    }) = expr
+    {
+        if let Expr::Name(ExprName { id, .. }) = func.as_ref() {
+            if id == "super" {
+                if let Some(first) = arguments.args.first() {
+                    return is_bad_super_call(first, current_class_name).map(|id| {
+                        Diagnostic::new(
+                            BadSuperCall {
+                                bad_super_arg: id.to_owned(),
+                            },
+                            range.to_owned(),
+                        )
+                    });
+                }
+            }
+        }
+        return check_expr(func, current_class_name);
+    } else if let Expr::Attribute(ExprAttribute { value, .. }) = expr {
+        return check_expr(value, current_class_name);
+    }
+    None
+}
+
+fn traverse_body(body: &[Stmt], current_class_name: &str) -> Vec<Diagnostic> {
```

Comment:
I think this will warn about

```python
class Test:
    a = super()
```

because `traverse_body` is used to traverse the class and nested function bodies. Is this intentional?

Reply:
I checked it and it doesn't.

Reply:
```cat bad_super.py
class Something:
    pass


class Test:
    def __init__(self):
        super = test


class O(Test):
    def __init__(self):
        super(Something, self).__init__()


class Test2:
    a = super()```

Reply:
```$ cargo run -p ruff -- check bad_super.py --no-cache --preview --select PLE1003
   Compiling ruff_linter v0.3.0 (/home/chris/projects/ruff/crates/ruff_linter)
   Compiling ruff_workspace v0.0.0 (/home/chris/projects/ruff/crates/ruff_workspace)
   Compiling ruff v0.3.0 (/home/chris/projects/ruff/crates/ruff)
    Finished dev [unoptimized + debuginfo] target(s) in 14.73s
     Running `target/debug/ruff check bad_super.py --no-cache --preview --select PLE1003`
bad_super.py:12:9: PLE1003 Bad first argument 'Something' given to super()
   |
10 | class O(Test):
11 |     def __init__(self):
12 |         super(Something, self).__init__()
   |         ^^^^^^^^^^^^^^^^^^^^^^ PLE1003
   |

Found 1 error.```